src/app/tip-sheets/pages/detail/detail.page.ts

```diff
@@ -0,0 +1,40 @@
+import { Component, OnDestroy } from "@angular/core";
+import { ActivatedRoute, Router } from "@angular/router";
+import { AppTranslateService } from "src/app/modules/translation.module";
+import { TIP_SHEETS } from "../../data/tip-sheets.data";
+
+@Component({
+  selector: "app-detail",
+  templateUrl: "./detail.page.html",
+  styleUrls: ["./detail.page.scss"],
+})
+export class TipSheetDetailPage implements OnDestroy {
+  totalTipSheets = TIP_SHEETS.length;
+  tipSheet: typeof TIP_SHEETS[0];
+  constructor(
+    private route: ActivatedRoute,
+    private router: Router,
+    private translations: AppTranslateService
+  ) {
+    const sheetNumber = Number(route.snapshot.params.tipSheetNumber);
+    const tipSheet = TIP_SHEETS.find((s) => s.number === sheetNumber);
+    this.translations
```

Comment:
I like using a service to change the translation source, I also really like that it returns a promise so we only populate translated templates once the translation has loaded.